WooCommerce/Classes/ViewRelated/Orders/Order Details/OrderDetailsViewController.swift

```diff
@@ -939,46 +940,50 @@ private extension OrderDetailsViewController {
         present(navigationController, animated: true)
     }
 
-    func setOrderStatus(to newStatus: OrderStatusEnum?) {
+    func setOrderStatus(to newStatus: OrderStatusEnum?, completionHandler: @escaping () -> Void) {
         guard let newStatus = newStatus else {
             return
         }
-        let orderID = viewModel.order.orderID
+        let order = viewModel.order
         let undoStatus = viewModel.order.status
-        let done = updateOrderStatusAction(siteID: viewModel.order.siteID, orderID: viewModel.order.orderID, status: newStatus)
-        let undo = updateOrderStatusAction(siteID: viewModel.order.siteID, orderID: viewModel.order.orderID, status: undoStatus)
+        let done = updateOrderStatusAction(siteID: order.siteID, order: order, status: newStatus, closeListView: completionHandler)
+        let undo = updateOrderStatusAction(siteID: order.siteID, order: order, status: undoStatus, closeListView: completionHandler)
 
         ServiceLocator.stores.dispatch(done)
 
         ServiceLocator.analytics.track(.orderStatusChange,
-                                       withProperties: ["id": orderID,
+                                       withProperties: ["id": order.orderID,
                                                         "from": undoStatus.rawValue,
                                                         "to": newStatus.rawValue])
-
+        // TODO: Move this to display once the list view dismisses
         displayOrderStatusUpdatedNotice {
             ServiceLocator.stores.dispatch(undo)
             ServiceLocator.analytics.track(.orderStatusChange,
-                                           withProperties: ["id": orderID,
+                                           withProperties: ["id": order.orderID,
                                                             "from": newStatus.rawValue,
                                                             "to": undoStatus.rawValue])
         }
     }
-
     /// Returns an Order Update Action that will result in the specified Order Status updated accordingly.
     ///
-    private func updateOrderStatusAction(siteID: Int64, orderID: Int64, status: OrderStatusEnum) -> Action {
-        return OrderAction.updateOrderStatus(siteID: siteID, orderID: orderID, status: status, onCompletion: { [weak self] error in
-            guard let error = error else {
+    private func updateOrderStatusAction(siteID: Int64, order: Order, status: OrderStatusEnum, closeListView: @escaping () -> Void) -> Action {
```

Comment:
```suggestion
    private func updateOrderStatusAction(siteID: Int64, order: Order, status: OrderStatusEnum, onCompletion: @escaping () -> Void) -> Action {
```

We should probably keep the convention and call these closures, `onCompletion`